Improve accessibility of empty anchor elements

PaulMorris opened the pull request:
Anchor elements with no inner HTML content are an accessibility problem.  This PR fixes cases of this by replacing the anchors with `<button>` elements where possible or else by moving background images into the anchor tags where that is a more straightforward solution (as with the search buttons).  Title, aria-label, and role="button" attributes are added as appropriate to improve accessibility.

Search buttons tested by:
- log in as p1, do a search from the quick search field in the main navigation bar
- log in as system, do a search from the quick search field in the main navigation bar

The search should happen as expected (search results may vary).

Close buttons `[X]` on modal dialogs tested by:
- log in as p1, click 'Save as Draft' from an enrollment step page, click the `[X]` at top right of modal
- log in as p1, click 'Provider Lookup' from enrollment step page, click the `[X]` at top right of modal
- log in as p1, click 'Submit Enrollment' from enrollment step page, click the `[X]` at top right of modal
- log in as system, click `Delete` for a user in the list, click the `[X]` at top right of modal

The modal should close as expected.

The close buttons in the enrollment steps are not currently exposed in the UI, so cannot be tested at this point.

Resolves #507 Fix anchors that have no link content

jasonaowen commented on psm-app/cms-web/WebContent/WEB-INF/pages/provider/enrollment/steps/pageTemplates/default/member_information.jsp:
```diff
@@ -262,7 +262,7 @@
     <div class="bl"></div>
     <div class="br"></div>
-    <a href="javascript:" class="closeSection"></a>
+    <button class="closeSection" title="Close" aria-label="Close"></button>
```
This JSP appears to be intended to be included inside of a `<form>`, judging by the various `<input>` elements. Do `<button>`s inside `<form>`s always cause the form to submit? Or do they have to be `type="submit"` for that to happen?

(Also, is this one of the JSPs that are never included anywhere?)

PaulMorris replied:
Glad you asked @jasonaowen .  I checked [MDN](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/button) and buttons in forms are `type="submit"` by default (even if the type is not specified) and submit the form.  To prevent them from submitting the form, specify `type="button"` instead.  So I should do that here (if we are going to use `<button>`).

And yes, this is one of the JSP's not currently included anywhere.

PaulMorris commented:
@jasonaowen , this version addresses your comment by adding `type="button"` to the close buttons in the enrollment steps JSPs that are not yet included anywhere.

Also, I realized that the search form for the system admin needed a submit `<button>` for a11y (rather than an anchor).*  I changed it to a `<button>`, then changed the non-system-admin one to a button as well to treat them the same with CSS (using background image rather than an `<img>` in the html).  But that broke the JS function call for non-system-admin search, because the `<button>` can't have an href.  So I refactored the JS function, which entailed making `ctx` accessible to the JS on provider pages.  And in the process re-organized the `html_head.template.html` file a bit.

* PR #542 landed on master after I'd done my inventory of these forms that need submit buttons, so there are a few of them that that PR introduced that still need fixing.  This was one of them.

PaulMorris commented:
It might be worth going ahead and putting the search box and button inside a `<form>` and using that rather than relying on the JS click handler (for the non-system admin pages).

cecilia-donnelly commented:
@jasonaowen, can you re-review this?  I'm going to go through your open PRs first, and will get to this after that.

jasonaowen commented on psm-app/cms-web/WebContent/templates/includes/nav.template.html:
```diff
@@ -53,7 +53,8 @@
           <div class="inputContainer">
-            <a id="enrollmentQuickSearch" href="{{ctx}}/provider/search/quick" class="search"></a> <input id="quickSearchInput" type="text" class="hint" value="{{#if param.npi }} {{param.npi}} {{else}} Search Keyword {{/if}}" title="Search Keyword" />
+            <button id="enrollmentQuickSearch" class="search" title="Search" aria-label="Search"></button>
+            <input id="quickSearchInput" type="text" class="hint" value="{{#if param.npi }} {{param.npi}} {{else}} Search Keyword {{/if}}" title="Search Keyword" />
```
Something I missed in reviewing #409 is that the spaces in between `{{else}}` and `{{/if}}` break the "hint"-y nature, which is currently implemented in JavaScript. I opened #612 for that; @PaulMorris, you can fix it in this PR or not, as you see fit.
